sklearn/utils/tests/test_sparsefuncs.py

```diff
@@ -522,3 +523,17 @@ def test_inplace_normalize():
                 if inplace_csr_row_normalize is inplace_csr_row_normalize_l2:
                     X_csr.data **= 2
                 assert_array_almost_equal(np.abs(X_csr).sum(axis=1), ones)
+
+
+def test_sparse_unique():
+    X_sparse = sp.csr_matrix(np.array([[1, 0, 0, 0],
+                                       [0, 1, 0, 0],
+                                       [0, 0, 0, 3],
+                                       [4, 0, 0, 0]], dtype=np.int8))
+    unique = sparse_unique(X_sparse)
+    assert_equal(unique, np.array([0, 1, 3, 4]))
+
+    # Degenerate case where one uses CSR/CSC matrices to store dense data
+    X_notsparse = sp.csr_matrix(np.array([1, 1, 1, 2, 3]))
+    unique2 = sparse_unique(X_notsparse)
+    assert_equal(unique2, np.array([1, 2, 3]))
```

Comment:
We don't compare arrays with `assert_equal`. You can either use `assert np.all(unique2 == np.array([1, 2, 3]))` or use `assert_array_equal`